### ISSUE TITLE
Data table extensions: column alignment

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-cloud-cognitive#1614]
# Data table extensions: column alignment

## Brief description

This issue will deliver the column alignment data table extension.

## Details

[Link to Design](https://pages.github.ibm.com/cdai-design/pal/components/data-table/column-alignment)

### Links to other materials

e.g. sketch files, icons, images, prototypes

## Design Maintainer(s)

- Marion Brülls

## Tasks

Before starting work on this epic, please review and complete the following.

- [ ] Initial review of design/existing code

### Working in Carbon for IBM Products package

- [ ] Have you reviewed our
      [contribution guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/.github/CONTRIBUTING.md)?
- [ ] Have you noted our
      [code guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/docs/CODE_GUIDELINES.md)?
- [ ] Did you use the CLI to scaffold your new component?

**Comment by elycheea** [comment]
Looks like a duplicate of #1925 so closing this one. We’ll refer to the newer issue.
